cmd/podman/images/pull.go

```diff
@@ -154,17 +159,28 @@ func imagePull(cmd *cobra.Command, args []string) error {
 	}
 	// Let's do all the remaining Yoga in the API to prevent us from
 	// scattering logic across (too) many parts of the code.
-	pullReport, err := registry.ImageEngine().Pull(registry.GetContext(), args[0], pullOptions.ImagePullOptions)
-	if err != nil {
-		return err
+	for _, arg := range args {
+		found, err := registry.ImageEngine().Exists(registry.GetContext(), arg)
+		if err != nil {
+			return err
+		}
+		if found.Value {
+			fmt.Printf("%s already exists\n", arg)
+		} else {
+			fmt.Print("\n")
+			_, err = registry.ImageEngine().Pull(registry.GetContext(), arg, pullOptions.ImagePullOptions)
+			if err != nil {
+				return err
+			}
+		}
 	}
 
-	if len(pullReport.Images) > 1 {
+	/*if len(pullReport.Images) > 1 {
```

Comment:
This needs finishing.

Reply:
> If the :latest tag on a repo changes, why wouldn’t podman pull verify that it has the updated version?

My bad, I didn't realize. Will fix that

Reply:
> This needs finishing.

I forgot to ask, I didn't understand what the portion of code I commented is for?

Reply:
@infiniteregrets For each pull, you need to print out the image ids that were pulled.

Reply:
@jwhonce I was a bit confused because, "Pulled Images:" was never getting printed before only the signature was. I have made the changes, please let me know if this is okay now?